src/withStyles.jsx

```diff
@@ -51,28 +51,47 @@ export function withStyles(
   /** Cache for storing the result of stylesFn(theme) for all themes. */
   const stylesFnResultCacheMap = typeof WeakMap === 'undefined' ? new Map() : new WeakMap();
 
+  /** Cache for storing the results of computations, theme -> WithStyles -> direction -> results */
```

Comment:
Could we write typescript type for clarity in the comment?

I think it's this:

```
Map<Theme, Map<typeof WithStyles, { ltr: {}, rtl: {} }>>
```